src/avram/associations.cr

```diff
@@ -165,7 +165,7 @@ module Avram::Associations
       def preload(preload_query : {{ model }}::BaseQuery)
         add_preload do |records|
           ids = records.map(&.id)
-          {{ assoc_name }} = preload_query.{{ foreign_key }}.in(ids).results.group_by(&.{{ foreign_key }})
+          {{ assoc_name }} = preload_query.dup.{{ foreign_key }}.in(ids).results.group_by(&.{{ foreign_key }})
```

Comment:
Here is the problem. We were add the `in(ids)` every time a preload proc was called. So the second time `results` was called, it would add another identical clause. This was a no-no.

Instead we duplicate the query so we can get a fresh copy of the query that we use just for preloading w/o modifying the original query.